server/services/list/service.js

```diff
@@ -173,24 +173,23 @@ const getInstance = (requestId, user) => {
 };
 
 const flush = async (key) => {
-    listCache.flush(key);
-    getLogger().info('Flushing Cache ' + key);
+    getLogger().info('Flushing and retrieving new Cache ' + key);
     await cacheStaticList(key);
 };
 
 const cacheStaticList = async (listId) => {
     const { endpoint, client, adapter } = listRepository.fetch(listId);
     const clientInstance = clientRepository.fetch(client);
     const logger = getLogger();
-    let response;
     try {
-        response = await clientInstance.get(endpoint);
+        const response = await clientInstance.get(endpoint);
+        listCache.flush(listId);
+        const listData = await applyAdapter(response.data, adapter, { logger });
```

Comment:
Worth moving this up a line, I doubt the applying of the adapter will take long, but would remove the ability to flush and then the adapter to fail...

Reply:
Good shout, I've made the change